Formula/dotnet.rb

```diff
@@ -20,31 +15,63 @@ class Dotnet < Formula
 
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
   depends_on xcode: :build
-  depends_on arch: :x86_64
-  depends_on "curl"
   depends_on "icu4c"
   depends_on "openssl@1.1"
 
+  # HACK: this should not be a test dependency but is due to a limitation with fails_with
+  uses_from_macos "llvm" => [:build, :test]
```

Comment:
**TODO**: Looks like Clang 13 will hit some issues in v6.0.100 based on Fedora package https://src.fedoraproject.org/rpms/dotnet6.0/blob/48eddbd9d70bd0c522e2b31ffbc43dd84a4f4f39/f/dotnet6.0.spec#_92

Can either patch (https://github.com/dotnet/runtime/pull/62170) or switch to different `llvm/gcc`. Latter may partly reduce required space since `llvm` has `gcc` dependency but older `llvm <= 12` are patched to work with older `gcc-5`.

Reply:
Homebrew's compiler logic fails when combining `fails_with :gcc` and `depends_on "llvm@..."`, so will need to patch source for Clang 13 support.